docs: update the configuration page (#2083)
unjs/nitro: docs/content/1.guide/1.configuration.md

# Configuration

> You can customize your Nitro server with single configuration file called `nitro.config.ts`. This file will be used by [deployement presets](/deploy) so you do not need to worry about them when configuring your server.

Reviewer: THe left senstence does not means that -- it meant that presets are **made with** same interface 😅 Let's remove it altogether for now seems more confusing

```suggestion
You can customize your Nitro server with a single configuration file `nitro.config.ts`.
```

## Runtime Configuration

Nitro provides a runtime config API to expose configuration within your application, with the ability to update it at runtime by setting environment variables.

To expose config and environment variables to the rest of your app, you will need to define runtime configuration in your configuration file.
Nitro provides a runtime config API to expose configuration within your application, with the ability to update it at runtime by setting environment variables. This is useful when you want to expose different configuration values for different environments (e.g. development, staging, production). For example, you can use this to expose different API endpoints for different environments or to expose different feature flags.

**Example:**
First, you need to define the runtime config in your configuration file.

::code-group
```ts [nitro.config.ts]
export default defineNitroConfig({
  runtimeConfig: {
    helloThere: "foobar",
    foo: "bar", // `bar` is the default value
  }
})
```

```ts [nuxt.config.ts]
export default defineNuxtConfig({
  runtimeConfig: {

>     helloThere: "foobar",

Reviewer: using `helloThere` were intentional to show multi part override via env (`NUXT_HELLO_THERE`) we might make it more realistic example like `apiToken` that still has two parts

Author: You are right. I've update it.

You can now access the runtime config using `useRuntimeConfig(event)`.  Use `useRuntimeConfig(event)` within event handlers and utilities and **avoid** calling it in ambient global contexts. This could lead to unexpected behavior such sharing the same runtime config across different requests.

::code-group
```ts [api/example.get.ts (nitro)]
export default defineEventHandler((event) => {

>   return useRuntimeConfig(event).apiToken // Return `dev_token`

Reviewer: ```suggestion
  return useRuntimeConfig(event).apiToken // Returns `dev_token`
```

});
```

```ts [server/api/example.get.ts (nuxt)]
export default defineEventHandler((event) => {
  return useRuntimeConfig(event).apiToken // dev_token
});
```
::

`useRuntimeConfig(event)` is an auto-imported function. Read more about [auto-imports](/auto-imports).

### Local development

Finally, you can update the runtime config using environment variables. You can use a `.env` file in development and use platform variables in production (see below).

Create an `.env` file in your project root:

```bash [.env]
API_TOKEN="123"
```

Re-start the development server, fetch the `/api/example` endpoint and you should see `123` as the response instead of `dev_token`.

> Do not froget taht you can still universally access environment variables using `import.meta.env` or `process.env` but avoid using them in ambiant global contexts to prevent unexpected behavior.

Reviewer: ```suggestion
Do not forget that you can still universally access environment variables using `import.meta.env` or `process.env` but avoid using them in ambiant global contexts to prevent unexpected behavior.
```
